src/components/step-indicator/StepIndicator.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import Animated, {
+  Easing,
+  useAnimatedStyle,
+  useDerivedValue,
+  withRepeat,
+  withSequence,
+  withTiming,
+} from 'react-native-reanimated';
+import { Box, Columns } from '@rainbow-me/design-system';
+
+const PULSE_STEP_DURATION = 1000;
+
+type StepIndicatorProps = {
+  steps: number; // 1 indexed
+  currentStep: number; // set higher than `steps` to complete the animation
+};
+
+export default function StepIndicator({
+  steps,
+  currentStep,
+}: StepIndicatorProps) {
+  const pulseStepOpacity = useDerivedValue(() =>
+    withRepeat(
+      withSequence(
+        withTiming(0, { duration: PULSE_STEP_DURATION }),
+        withTiming(1, { duration: PULSE_STEP_DURATION }),
+        withTiming(0, { duration: PULSE_STEP_DURATION })
+      ),
+      -1
+    )
+  );
+
+  const pulseStepTranslate = useDerivedValue(() =>
+    withRepeat(
+      withSequence(
+        withTiming(-100, { duration: PULSE_STEP_DURATION }),
+        withTiming(-100, { duration: PULSE_STEP_DURATION }),
+        withTiming(0, { duration: PULSE_STEP_DURATION })
+      ),
+      -1
+    )
+  );
+
+  const finishedStepFill = useDerivedValue(() =>
+    withSequence(
+      withTiming(0, { duration: 300, easing: Easing.out(Easing.ease) }),
+      withTiming(100, { duration: 300, easing: Easing.out(Easing.ease) })
+    )
+  );
+
+  const animatedPulseStyle = useAnimatedStyle(() => ({
+    left: `${pulseStepTranslate.value}%`,
+    opacity: pulseStepOpacity.value,
+  }));
+
+  const animatedFinishedStyle = useAnimatedStyle(() => ({
+    width: `${finishedStepFill.value}%`,
+  }));
+
+  return (
+    <Box padding="10px" width="full">
+      <Columns space={{ custom: 9 }}>
+        {Array.from({ length: steps }).map((_, index) => {
+          const stepIndex = index + 1;
+          const isCurrentStep = stepIndex === currentStep;
+          const isFinished = currentStep > stepIndex;
+          const isPulsing = isCurrentStep && !isFinished;
+          const isAnimatingFill = stepIndex === currentStep - 1 && isFinished;
+
+          return (
+            <Box
+              borderRadius={16}
+              height={{ custom: 4 }}
+              key={index}
+              style={{
+                backgroundColor: 'hsla(261, 55%, 65%, 0.1)',
+                overflow: 'hidden',
+              }}
+              width="full"
+            >
+              {isPulsing ? (
+                <Box
+                  as={Animated.View}
+                  borderRadius={16}
+                  height={{ custom: 4 }}
+                  style={[
+                    {
+                      backgroundColor: 'hsla(261, 55%, 65%, 1)',
```

Comment:
Can we make these colors take the accent color instead?

Reply:
@jxom Yeah but the colors in Figma are different than the ones in the design system last time I checked.